node/full.go

```diff
@@ -313,6 +319,70 @@ func (n *FullNode) AppClient() abciclient.Client {
 	return n.appClient
 }
 
+type ResultDirectTx struct {
+	Included bool   `json:"included"`
+	Height   uint64 `json:"height"`
+}
+
+func (n *FullNode) ReceiveDirectTx() func([]byte) ResultDirectTx {
+	// returns "true" if a block is built, "false" if it times out.
+	return func(tx []byte) ResultDirectTx {
+		n.Logger.Info("Receive Direct Tx")
+		n.incomingTxCh <- tx
+		t := time.NewTimer(2 * time.Second)
+
+		// Non-progressive sequencer doesn't track
+		// whether or not the tx was included in a block
+		if !n.conf.ProgressiveSequencer {
+			return ResultDirectTx{
+				true,
+				0,
+			}
+		}
+
+		for {
+			select {
+			case <-t.C:
+				return ResultDirectTx{
+					false,
+					0,
+				}
+			case <-n.DoneBuildingBlock:
+				return ResultDirectTx{
```

Comment:
who is consuming the result? i don't think it is needed. the progressive sequencer node itself can log the block height at which the direct txns are included and this RPC can simply be POST kind, without requiring result.

Reply:
I think it will come in handy for facilitating a smooth UX for the end-users of these rollups

Reply:
I envision the clients sending transactions as requests, and receiving blocks with the updated state as a response, much like how it works with Web2 apps